src/data-workspace/validation/validation-config.js

```diff
@@ -0,0 +1,34 @@
+import i18n from '@dhis2/d2-i18n'
+import {
+    IconErrorFilled16,
+    IconWarningFilled16,
+    IconInfo16,
+    IconErrorFilled24,
+    IconWarningFilled24,
+    IconInfo24,
+} from '@dhis2/ui'
+
+export const validationLevels = ['HIGH', 'MEDIUM', 'LOW']
+export const validationLevelsConfig = {
```

Comment:
I think these should be components instead of config, that'd be more "react-ish".
Configs have the nasty issue that when there's an exception (like an edge case for a particular config), that exception has to be build into the general handling of the config rather than into the component that has to handle the exception.

Reply:
This requires a somewhat different approach how to compose things, e.g. instead of having a single validation priority group component, there would be three (`ValidationGroupHighPriority`, `ValidationGroupMediumPriority` & `ValidationGroupLowPriority`).

On the one hand this would create a bit more boiler plate code but on the other hand it's reduce complexity and increases maintainability as changing one of these composed components wouldn't affect the complexity of the others. Everything that's shared between them can be abstracted into its own component / function.

If you want to, we can talk about this in a call / on slack a bit more in case you disagree or have questions.

Reply:
hmm I disagree with this .. I don't think having `ValidationGroupHighPriority` `ValidationGroupMediumPriority` etc.. would reduce complexity nor is it a more "react-y" way of achieving what we want. The `importance` (high, medium or low) is a property of the component, and react way is to use `props` for this. We have a similar concept in many places in our component library, so for things like `NoticeBox`, we wouldn't have an `ErrorNoticeBox` and `WarningNoticeBox`, it's just a prop. I don't think that having a config (or a helper function) is too complex to justify duplication. And if we try to avoid that duplication by abstracting it into its own component, then we're just shifting the complexity into that shared component (it also feels a bit inheritance-like rather componentised). 

For me it helps to look at the design and its language, [the design specs](https://docs.google.com/document/d/1vS1urvJRePRugFqrPkfOEjpkbij1AfTz0BYp6RKV8LU/edit#heading=h.9xf56tvc4qp8) talk about `PriortyGroupUnit` as a component with some `titles` and `icons` for each scenario. Design changes would happen at that component level, not a particular instance of it. That's the natural level of abstraction imo, everything else is a prop to the component.

Now maybe a config is not the best way to save that map of props .. but then we could just change the config to a helper function.

Reply:
As discussed, we'll defer that, I need to prepare this thought in a proper manner so we can properly talk about it. This topic is highly subjective and depends on past experiences, so I'll try to show how I ended up with my position. You can leave the code as is